Tests/GRPCTests/AsyncClientCallTests.swift

```diff
@@ -0,0 +1,232 @@
+        path: "/echo.Echo/Update",
+        callOptions: .init()
+      )
+
+    // Spin up a task to send the requests with a delay before each one
+    Task {
+      let delay = TimeAmount.milliseconds(500)
+      for word in ["foo", "bar", "baz"] {
+        try await Task.sleep(nanoseconds: UInt64(delay.nanoseconds))
+        try await update.sendMessage(.with { $0.text = word })
+      }
+      try await Task.sleep(nanoseconds: UInt64(delay.nanoseconds))
+      try await update.sendEnd()
+    }
+
+    // ...and then wait on the responses...
+    var numResponses = 0
+    for try await _ in update.responseStream {
+      numResponses += 1
+    }
+
+    await assertThat(numResponses, .is(.equalTo(3)))
+    await assertThat(await update.status, .hasCode(.ok))
+  } }
+
+  func testAsyncBidirectionalStreamingCall_ConcurrentTasks() throws { XCTAsyncTest {
+    let channel = try self.setUpServerAndChannel()
+    let update: AsyncBidirectionalStreamingCall<Echo_EchoRequest, Echo_EchoResponse> = channel
+      .makeAsyncBidirectionalStreamingCall(
+        path: "/echo.Echo/Update",
+        callOptions: .init()
+      )
+
+    actor TestResults {
+      static var numResponses = 0
+      static var numRequests = 0
+    }
+
+    // Send the requests and get responses in separate concurrent tasks and await the group.
+    _ = await withThrowingTaskGroup(of: Void.self) { taskGroup in
+      // Send requests in a task, sleeping in between, then send end.
+      taskGroup.addTask {
+        let delay = TimeAmount.milliseconds(500)
+        for word in ["boyle", "jeffers", "holt"] {
+          try await Task.sleep(nanoseconds: UInt64(delay.nanoseconds))
+          try await update.sendMessage(.with { $0.text = word })
+          TestResults.numRequests += 1
+        }
+        try await Task.sleep(nanoseconds: UInt64(delay.nanoseconds))
+        try await update.sendEnd()
+      }
+      // Get responses in a separate task.
+      taskGroup.addTask {
+        for try await _ in update.responseStream {
+          TestResults.numResponses += 1
+        }
+      }
+    }
+    await assertThat(TestResults.numRequests, .is(.equalTo(3)))
+    await assertThat(TestResults.numResponses, .is(.equalTo(3)))
+    await assertThat(await update.status, .hasCode(.ok))
+  } }
+}
+
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+public extension XCTestCase {
+  /// Cross-platform XCTest support for async-await tests.
+  ///
+  /// Currently the Linux implementation of XCTest doesn't have async-await support.
+  /// Until it does, we make use of this shim which uses a detached `Task` along with
+  /// `XCTest.wait(for:timeout:)` to wrap the operation.
+  ///
+  /// - NOTE: Support for Linux is tracked by https://bugs.swift.org/browse/SR-14403.
+  /// - NOTE: Implementation currently in progress: https://github.com/apple/swift-corelibs-xctest/pull/326
+  func XCTAsyncTest(
+    expectationDescription: String = "Async operation",
+    timeout: TimeInterval = 3,
```

Comment:
Can we have a much longer default timeout? Say 30 seconds? CI is often slow so if we time out I'd rather be sure we're really wedged rather than just slow.